packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -493,24 +509,83 @@ class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>>
       controller: _optionsViewController,
       overlayChildBuilder: _buildOptionsView,
       child: TextFieldTapRegion(
-        child: Container(
-          key: _fieldKey,
-          child: Shortcuts(
-            shortcuts: _shortcuts,
-            child: Actions(
-              actions: _actionMap,
-              child: CompositedTransformTarget(
-                link: _optionsLayerLink,
-                child: fieldView,
+        // This LayoutBuilder ensures that _fieldBoxConstraints is assigned
```

Comment:
If the options view is also in a layout builder I don't think the post-frame callback is needed?

Reply:
I get the "setState called during build" error if I remove it.

Reply:
I think in that case the setState isn't necessary either. If the overlay portal and its child get a different BoxConstraints, the overlay child will relayout too. And the layout builder will rebuild. I think that happens in the same frame so it would work better than using a post frame callback.

Reply:
Hmm it doesn't seem to work when I remove that. When the constraints change, the LayoutBuilder in OverlayPortal.child is called with the new constraints, but the overlayChildBuilder is not called.

Reply:
I was wrong about using a layout builder. If `child` gets a different set of constraints, the layout builder callback will be called with the new constraints. But overlay child is laid out using the Overlay's constraints so its layout builder won't be called.